File: backend/entities/message.go
package entities

import "github.com/mochi-yu/websocket-practice/ent"

type MessageRequest struct {
	Content    string `json:"content"`
	AuthorName string `json:"author_name"`
}

func (mr *MessageRequest) ToEntMessage() (dest *ent.Message) {
	dest = &ent.Message{
		Content:    mr.Content,
		AuthorName: mr.AuthorName,
	}

	return
}

type MessageResponse struct {
	ID         int    `json:"id"`
	Content    string `json:"content"`
	AuthorName string `json:"author_name"`
	CreatedAt  string `json:"created_at"`
}

File: frontend/src/types/message.ts
export type TMessageRequest = {
  content: string
  author_name: string
}

export type TMessageResponse = {
  id: string
  content: string
  author_name: string
  created_at: string
}

export const ValidateMessageRequest = (req: TMessageRequest) => {
  if (req.content.length === 0) {
    return 'メッセージを入力してください'
  }
  if (req.author_name.length === 0) {
    return '名前を入力してください'
  }
}
